src/common/drain/drain.js

```diff
@@ -86,29 +87,33 @@ function checkCanDrainAll (agentIdentifier) {
    * the subscribed handlers for the group.
    * @param {*} group - The name of a particular feature's event "bucket".
    */
-function drainGroup (agentIdentifier, group) {
+function drainGroup (agentIdentifier, group, activateGroup = true) {
   const baseEE = agentIdentifier ? ee.get(agentIdentifier) : ee
   const handlers = defaultRegister.handlers // other storage in registerHandler
   if (!baseEE.backlog || !handlers) return
 
-  var bufferedEventsInGroup = baseEE.backlog[group]
-  var groupHandlers = handlers[group] // each group in the registerHandler storage
-  if (groupHandlers) {
-    // We don't cache the length of the buffer while looping because events might still be added while processing.
-    for (var i = 0; bufferedEventsInGroup && i < bufferedEventsInGroup.length; ++i) { // eslint-disable-line no-unmodified-loop-condition
-      emitEvent(bufferedEventsInGroup[i], groupHandlers)
-    }
+  // Only activated features being drained should run queued listeners on buffered events. Deactivated features only need to release memory.
+  if (activateGroup) {
+    const bufferedEventsInGroup = baseEE.backlog[group]
+    const groupHandlers = handlers[group] // each group in the registerHandler storage
+    if (groupHandlers) {
+      // We don't cache the length of the buffer while looping because events might still be added while processing.
+      for (let i = 0; bufferedEventsInGroup && i < bufferedEventsInGroup.length; ++i) { // eslint-disable-line no-unmodified-loop-condition
+        emitEvent(bufferedEventsInGroup[i], groupHandlers)
+      }
 
-    mapOwn(groupHandlers, function (eventType, handlerRegistrationList) {
-      mapOwn(handlerRegistrationList, function (i, registration) {
-        // registration is an array of: [targetEE, eventHandler]
-        registration[0].on(eventType, registration[1])
+      mapOwn(groupHandlers, function (eventType, handlerRegistrationList) {
+        mapOwn(handlerRegistrationList, function (i, registration) {
+          // registration is an array of: [targetEE, eventHandler]
+          registration[0].on(eventType, registration[1])
+        })
       })
-    })
+    }
   }
+
   if (!baseEE.isolatedBacklog) delete handlers[group]
   baseEE.backlog[group] = null
-  baseEE.emit('drain-' + group, [])
+  baseEE.emit('drain-' + group, []) // exists purely for a unit test
```

Comment:
Can we make this comment a TODO so we can look at removing it. Don't like having code around that is purely for testing.

```suggestion
  baseEE.emit('drain-' + group, []) // TODO: Code exists purely for a unit test and needs to be refined
```